include/ppx/fs.h

```diff
@@ -46,27 +46,69 @@ namespace ppx::fs {
 void set_android_context(android_app* androidContext);
 #endif
 
-//! @class File
-//!
-//!
+// Abstract a static, regular file on all platforms.
+// This class doesn't handle sockets, or file which content is not constant
+// for the lifetime of this class.
 class File
 {
+    // Internal enum to know which handle is currently valid.
+    enum FileHandleType
+    {
+        // Default handle, no file associated.
+        BAD_HANDLE = 0,
+        // The file is accessible throught a stream.
+        STREAM_HANDLE = 1,
+        // The file is accessible through an Android asset handle.
+        ASSET_HANDLE = 3
+    };
+
 public:
-    bool   Open(const std::filesystem::path& path);
-    bool   IsOpen() const;
-    size_t Read(void* buf, size_t count);
-    size_t GetLength();
-    void   Close();
-#if defined(PPX_ANDROID)
-    const void* GetBuffer();
-#endif
+    File();
+    File(const File& other)  = delete;
+    File(const File&& other) = delete;
+    ~File();
+
+    // Open a file given a specific path.
+    // path: the path of the file to open.
+    //  - On desktop, loads the regular file at `path`. (memory-mapping availability is implementation defined).
+    //  - On Android, relative path are assumed to be loaded from the APK, those are memory mapped.
+    //                absolute path are loaded as regular files (mapping availability is implementation defined).
```

Comment:
Do we need to do anything special to read those? For example add READ_EXTERNAL_STORAGE to manifest?

Reply:
Looks like this will depend on the data location?
I just tried loading the CubeTextured's texture from `/data/local/tmp/` instead of the APK, and it works with checked-in manifests.